cli/src/pcluster/resources/batch/docker/scripts/modify_yum_vars.sh

```diff
@@ -41,14 +41,22 @@ main() {
     fi
 
     # Only modify yum repo variables if cluster is running in a China region.
-    REGION="${1}"
-    if [[ "${REGION}" =~ ^cn-.* ]]; then
+    local REGION="${1}"
+    local AWS_DOMAIN="amazonaws.com"
+    if [[ ${REGION} == cn-* ]]; then
+        AWS_DOMAIN="amazonaws.com.cn"
+    elif [[ ${REGION} == us-iso-* ]]; then
+        AWS_DOMAIN="c2s.ic.gov"
+    elif [[ ${REGION} == us-isob-* ]]; then
+        AWS_DOMAIN="sc2s.sgov.gov"
+    fi
+    if [[ ${REGION} == cn-* ]] || [[ ${REGION} == us-iso-* ]] || [[ ${REGION} == us-isob-* ]]; then
```

Comment:
instead of duplicating the comparison logic here, why not just check if AWS_DOMAIN is different from the default then enter the if